gfsphysics/GFS_layer/GFS_diagnostics.F90

```diff
@@ -2307,173 +2307,494 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%du3dt(:,:,5)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'dv3dt_damp'
+      ExtDiag(idx)%desc = 'v momentum tendency due to Rayleigh damping'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%dv3dt(:,:,5)
+      enddo
+
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'du3dt_shalcnv'
+      ExtDiag(idx)%desc = 'u momentum tendency due to shallow convection'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%du3dt(:,:,6)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'dv3dt_shalcnv'
+      ExtDiag(idx)%desc = 'v momentum tendency due to shallow convection'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%dv3dt(:,:,6)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'du3dt_phys'
+      ExtDiag(idx)%desc = 'u momentum tendency due to physics'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%du3dt(:,:,7)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'dv3dt_phys'
+      ExtDiag(idx)%desc = 'v momentum tendency due to physics'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%dv3dt(:,:,7)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'du3dt_nophys'
+      ExtDiag(idx)%desc = 'u momentum tendency due to non-physics processes'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
+      ExtDiag(idx)%time_avg = .TRUE.
+      allocate (ExtDiag(idx)%data(nblks))
+      do nb = 1,nblks
+         ExtDiag(idx)%data(nb)%var3 => IntDiag(nb)%du3dt(:,:,8)
+      enddo
+
+      idx = idx + 1
+      ExtDiag(idx)%axes = 3
+      ExtDiag(idx)%name = 'dv3dt_nophys'
+      ExtDiag(idx)%desc = 'v momentum tendency due to non-physics processes'
+      ExtDiag(idx)%unit = 'm s-2'
+      ExtDiag(idx)%mod_name = 'gfs_phys'
```

Comment:
for tendency from non-physics processes, maybe it's good not to put it in "gfs_phys" model in the future

Reply:
Good point. The only reason why it ended up there was "convenience" so that all tendencies were in the same file. We can easily change this.